phpBB/includes/functions_content.php

```diff
@@ -400,16 +396,14 @@ function get_context(string $text, array $words, int $length = 400): string
 		$fragment_end = $end - $start + 1;
 
 		// Find the first valid alphanumeric character in the fragment to don't cut words
-		if ($start > 0)
+		if ($start > 0 && preg_match('/[^a-zA-Z0-9][a-zA-Z0-9]/u', $fragment, $matches, PREG_OFFSET_CAPTURE))
```

Comment:
This can be changed to the next character to a space, point, comma, etc to be valid in other languages

Reply:
I don't really know how would it look like, could you please suggest a code change for that?

Reply:
Something like
```php
if ($start > 0 && preg_match('/[\s!?.,]+(.)/u', $fragment, $matches, PREG_OFFSET_CAPTURE))
```

Reply:
I'd probably better just convert current regex to unicode which should yield the same result but for any language, like `/[^\p{L}\p{N}][\p{L}\p{N}]/ui`.

Reply:
Much better, I didn't know that regexp pattern